recipe/meta.yaml

```diff
@@ -1,123 +1,206 @@
-    # The cuda-python package supports CUDA Enhanced Compatibility through its
-    # use of dlopen, so it is compatible with other CUDA minor versions.
-    - cuda-cudart-dev
-    - cuda-nvrtc-dev
+  number: 0
+  skip: true  # [py < 39 or ppc64le or osx]
 
+# Have certain top-level requirements so conda-smithy can render the correct variants
 requirements:
   build:
     - {{ compiler('c') }}
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != "None"]
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
     - {{ stdlib('c') }}
-    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
-    - python                              # [build_platform != target_platform]
-    - cython                              # [build_platform != target_platform]
-    - cuda-version {{ major_version }}.{{ minor_version }}
   host:
-    - cuda-version {{ major_version }}.{{ minor_version }}
-    - cuda-cudart-dev
-    - cuda-cudart
-    - cuda-nvrtc-dev
-    - cuda-nvrtc
-    - cuda-profiler-api
-    - cython
-    - pip
-    - pyclibrary
-    - python
-    - setuptools
-    # TODO: remove these in the next release
-    - tomli           # [py<311]
-    - versioneer
-  run:
     - python
-    - {{ pin_compatible('cuda-version', min_pin='x', max_pin='x') }}
-    # cuda-python requires cuda-nvrtc from the same major version. We ignored
-    # run-exports of cuda-nvrtc-dev and instead allow a looser pinning here.
-    - {{ pin_compatible('cuda-nvrtc', min_pin='x', max_pin='x') }}
-    - pywin32  # [win64]
-  run_constrained:
-    # cuda-cudart is not a hard dependency because cuda-python re-implements
-    # cudart. This package is optionally dlopen'd for getting the local cudart
-    # version with cuda.cudart.getLocalRuntimeVersion(). We ignored run-exports
-    # of cuda-cudart-dev and instead allow a looser pinning here.
-    - {{ pin_compatible('cuda-cudart', min_pin='x', max_pin='x') }}
 
-test:
-  requires:
-    - cuda-nvcc
-    - cuda-nvrtc-dev
-    - cuda-profiler-api
-    - cython
-    - setuptools
-  files:
-    - test_cimport.pyx
-  commands:
-    - export C_INCLUDE_PATH="$CONDA_PREFIX/{{ subdir }}/include"  # [linux]
-    - set "CL=%CL% /I%CONDA_PREFIX%\{{ subdir }}\include"         # [win]
-    - cythonize -i -3 test_cimport.pyx
-    - python -c "import test_cimport; test_cimport.test()"
-    - python -c "import cuda.bindings; print(cuda.bindings.__version__)"
-    - python -c "import cuda; print(cuda.__version__)"  # TODO: Remove this
-  imports:
-    - cuda
-    - cuda.cuda
-    - cuda.cudart
+
+outputs:
+
+  {% if building_bindings %}
+  - name: cuda-python
+    version: {{ pkg_ver }}
+    script: build_bindings.sh   # [linux]
+    script: build_bindings.bat  # [win]
+    build:
+      number: {{ number }}
+      # cuda.bindings needs to be built with the matching major.minor headers
+      {% if not (environ.get("cuda_compiler_version")|string()) == (major_version ~ "." ~ minor_version) %}
```

Comment:
Note: This skip condition has been updated to ensure the requirement is visible in the recipe level, not in the private migrator level. In fact, the migrator has caused trouble during rerendering, so I simply removed it. I checked that the new recipe here builds fine (if we revert the deselection logic to build `cuda-python`).